generators/upgrade/index.js

```diff
@@ -109,24 +132,42 @@ module.exports = UpgradeGenerator.extend({
         detectCurrentBranch: function() {
             var done = this.async();
             this.gitExec(['rev-parse', '-q', '--abbrev-ref', 'HEAD'], function(code, msg, err) {
-                if (code !== 0) this.error('Unable to detect current git branch:\n' + err);
+                if (code !== 0) this.error('Unable to detect current git branch:\n' + msg + ' ' + err);
                 this.sourceBranch = msg.replace('\n','');
                 done();
             }.bind(this));
         },
 
         prepareUpgradeBranch: function() {
             var done = this.async();
-            var createUpgradeBranch = function(callback) {
-                this.gitExec(['branch', '-q', UPGRADE_BRANCH], function(code, msg, err) {
-                    if (code !== 0) this.error('Unable to create ' + UPGRADE_BRANCH + ':\n' + err);
+            var recordCodeHasBeenGenerated = function() {
+                this.gitExec(['merge', '--strategy=ours', '-q', '--no-edit', UPGRADE_BRANCH], function(code, msg, err) {
```

Comment:
yeah you are right, that would have been easier.. I just explained the reason why in the issue : https://github.com/jhipster/generator-jhipster/issues/3696